lib/ansible/modules/cloud/digital_ocean/digital_ocean_image_facts.py

```diff
@@ -0,0 +1,170 @@
+    They both refer to the v2 token.
+
+requirements:
+  - "python >= 2.6"
+'''
+
+
+EXAMPLES = '''
+- name: Gather facts about all images
+  digital_ocean_image_facts:
+    image_type: all
+    oauth_token: "{{ oauth_token }}"
+
+- name: Gather facts about application images
+  digital_ocean_image_facts:
+    image_type: application
+    oauth_token: "{{ oauth_token }}"
+
+- name: Gather facts about distribution images
+  digital_ocean_image_facts:
+    image_type: distribution
+    oauth_token: "{{ oauth_token }}"
+
+'''
+
+
+RETURN = '''
+data:
+    description: DigitalOcean image facts
+    returned: success
+    type: list
+    sample: [
+        {
+            "created_at": "2018-02-02T07:11:43Z",
+            "distribution": "CoreOS",
+            "id": 31434061,
+            "min_disk_size": 20,
+            "name": "1662.1.0 (beta)",
+            "public": true,
+            "regions": [
+                "nyc1",
+                "sfo1",
+                "nyc2",
+                "ams2",
+                "sgp1",
+                "lon1",
+                "nyc3",
+                "ams3",
+                "fra1",
+                "tor1",
+                "sfo2",
+                "blr1"
+            ],
+            "size_gigabytes": 0.42,
+            "slug": "coreos-beta",
+            "type": "snapshot"
+        },
+    ]
+'''
+
+from traceback import format_exc
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.digital_ocean import DigitalOceanHelper
+from ansible.module_utils._text import to_native
+
+
+def core(module):
+    image_type = module.params['image_type']
+
+    rest = DigitalOceanHelper(module)
+
+    # Check if api_token is valid or not
+    response = rest.get('account')
+    if response.status_code == 401:
+        module.fail_json(msg='Failed to login using api_token, please verify '
+                             'validity of api_token')
+    base_url = 'images?'
+    if image_type == 'distribution':
+        base_url += "type=distribution&"
+    elif image_type == 'application':
+        base_url += "type=application&"
+    elif image_type == 'private':
+        base_url += "private=true&"
+
+    page = 1
+    has_next = True
+    images = []
+    status_code = 400
+    while has_next or status_code != 200:
+        required_url = "{0}page={1}&per_page=20".format(base_url, page)
+        response = rest.get(required_url)
+        status_code = response.status_code
+        # stop if any error during pagination
+        if 200 != status_code:
+            break
+        page += 1
+        images.extend(response.json["images"])
+        has_next = "pages" in response.json["links"] and "next" in response.json["links"]["pages"]
+
+    if status_code != 200:
+        module.fail_json(msg="Failed to fetch images facts due to error : %s" % response.json['message'])
+
+    module.exit_json(changed=False, data=images)
+
+
+def main():
+    module = AnsibleModule(
```

Comment:
A few days ago **lib/ansible/module_utils/digital_ocean.py** (Ansible Digital Ocean utils) was updated in PR #35934 which introduced generic configurations and documentation that should be used, particularly here this function **digital_ocean_argument_spec** (https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/digital_ocean.py)